lang/java/avro/src/main/java/org/apache/avro/file/DataFileWriter.java

```diff
@@ -190,11 +190,17 @@ public boolean isFlushOnEveryBlock() {
 
   /** Open a writer appending to an existing file. */
   public DataFileWriter<D> appendTo(File file) throws IOException {
-    return appendTo(new SeekableFileInput(file),
-                    new SyncableFileOutputStream(file, true));
+    SeekableInput input = new SeekableFileInput(file);
+    OutputStream output = new SyncableFileOutputStream(file, true);
```

Comment:
I think this line should be inside the try block as well to guarantee that input is always closed. Or, alternatively, it may be before the first line if order does not matter.